### PR TITLE
Re-associating route tables when ManagementEIP is required + other small changes

### PR DESCRIPTION
**Pull request by sumanth-lingappa**
*Issue #, if available:*

*Description of changes:*
**Major changes**
- Lambda ENI is moved to ServerSubnets from Management subnets
- As we were de-associating NATGw route-rule for server-side-route-tables in `DeleteServersideNATGWSRouteStack`, I have added required `DependsOn` and created a Delete scenario in the `DeleteRouteFunction` lambda function .
- New Security Group dedicated to lambda functions is created
- Made NATGw dis-association from server-side route tables as user-input param
+ some minor changes

**Edit:**
Earlier issues below (now resolved)

When `ManagementEIP` is required by the users, they cannot directly login to Citrix ADC as the management subnets are associated with Private Subnets. 
This PR checks if the user has requested `ManagementEIP`, and if yes, then it re-associates both primary and secondary management subnets to `PublicRouteTable`

Note:
> This PR is not complete yet. It is getting stuck at the lambda function because of the below error:
```
Exception: Connect timeout on endpoint URL: "https://ec2.ap-south-1.amazonaws.com/"
```
This happens when I re-associate subnets to PublicSubnets
@vsnyc : any reason for this behaviour?

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/citrix-adc-vpx-main.template.yaml [hunk 1]
```diff
@@ -538,7 +548,23 @@ Resources:
...
+      RouteTableId: !GetAtt VPCStack.Outputs.PublicSubnetRouteTable
+
+  SecondaryManagementSubnetRouteTabeleReassociation:
+    Condition: UseManagementEIP
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      SubnetId: !GetAtt VPCStack.Outputs.PrivateSubnet2AID
+      RouteTableId: !GetAtt VPCStack.Outputs.PublicSubnetRouteTable
+
+  DeleteServersideNATGWSRouteStack:
```
Comment (sumanth-lingappa) [inline templates/citrix-adc-vpx-main.template.yaml:565]: This stack deletes the NATGw association from server-subnet route-table.
If the user requests for NATGw to be associated, the he selects 'Yes', then this stack will be skipped.

File: templates/citrix-adc-vpx-main.template.yaml [hunk 2]
```diff
@@ -499,20 +501,28 @@ Parameters:
...
+      - "No"
+      - "Yes"
 
 Conditions:
   UsingDefaultBucket: !Equals [!Ref QSS3BucketName, "aws-quickstart"]
   CreateIAMRole: !Equals [!Ref CitrixADCIAMRole, ""]
   UseClientEIP: !Equals [!Ref ClientEIPRequired, "Yes"]
   UseManagementEIP: !Equals [!Ref ManagementEIPRequired, "Yes"]
   UseBastion: !Equals [!Ref BastionRequired, "Yes"]
+  NATGwOnServerSubnetCondition: !Equals [!Ref NATGwOnServersubnetRequired, "No"]
```
[the remark this answers is not in the document]
Comment (sumanth-lingappa) [reply templates/citrix-adc-vpx-main.template.yaml:518]: Yes, this condition is correct @vsnyc 
Idea is to skip `DeleteServersideNATGWSRouteStack` stack when the `NATGwOnServersubnetRequired` parameter is selected as `Yes`

> Please let me know if the condition and parameter is confusing.

**TLDR;**
- `VPCStack` stack creates NATGw association for all the private subnets.
- `DeleteServersideNATGWSRouteStack` stack deletes NATGw association from server-subnet-route-table.
However, if user wants to retain the NATGw association, he will select `Yes` to `NATGwOnServersubnetRequired ` parameter
